perfkitbenchmarker/scripts/execute_command.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -*- coding: utf-8 -*-
+"""Runs a command, saving stdout, stderr, and the return code in files.
+
+Simplifies executing long-running commands on a remote host.
+The status file (as specified by --status) will be exclusively locked until the
+child process exits.
+To await completion, "wait_for_command.py" acquires a shared lock on the
+status file, which blocks until the process completes.
+
+*This script runs on the remote VM*.
+"""
+import fcntl
+import logging
+import optparse
+import sys
+import subprocess
+
+
+def main():
+  p = optparse.OptionParser()
+  p.add_option('-o', '--stdout', dest='stdout', metavar='FILE',
+               help="""Write stdout to FILE. Required.""")
+  p.add_option('-e', '--stderr', dest='stderr', metavar='FILE',
+               help="""Write stderr to FILE.  Required.""")
+  p.add_option('-p', '--pid', dest='pid', help="""Write PID to FILE.""",
+               metavar='FILE')
+  p.add_option('-s', '--status', dest='status', help="""Write process exit
+               status to FILE. FILE with have an exclusive lock during run.
```

Comment:
quickly describe lock life cycle, e.g, when is the lock acquired, released (list all the conditions, especially those error conditions: process crashed/hang, etc)

Either in this comment here, or above.